geowebcache/core/src/main/java/org/geowebcache/storage/CompositeBlobStore.java

```diff
@@ -0,0 +1,360 @@
+        if (store == null) {
+            throw new StorageException("No default BlobStore has been defined");
+        }
+        return store;
+    }
+
+    /**
+     * Loads the blob stores from the list of configuration objects
+     * 
+     * @param configs the list of blob store configurations
+     * @return a mapping of blob store id to {@link LiveStore} containing the configuration itself
+     *         and the live instance if the blob store is enabled
+     * @throws ConfigurationException if there's a configuration error like a store confing having
+     *         no id, or two store configs having the same id, or more than one store config being
+     *         marked as the default one, or the default store is not
+     *         {@link BlobStoreConfig#isEnabled() enabled}
+     * @throws StorageException if the live {@code BlobStore} instance can't be
+     *         {@link BlobStoreConfig#createInstance() created} of an enabled
+     *         {@link BlobStoreConfig}
+     */
+    Map<String, LiveStore> loadBlobStores(List<? extends BlobStoreConfig> configs)
+            throws StorageException, ConfigurationException {
+
+        Map<String, LiveStore> stores = new HashMap<>();
+
+        BlobStoreConfig defaultStore = null;
+
+        try {
+            for (BlobStoreConfig config : configs) {
+                final String id = config.getId();
+                final boolean enabled = config.isEnabled();
+                if (Strings.isNullOrEmpty(id)) {
+                    throw new ConfigurationException("No id provided for blob store " + config);
+                }
+                if (stores.containsKey(id)) {
+                    throw new ConfigurationException("Duplicate blob store id: " + id
+                            + ". Check your configuration.");
+                }
+                if (CompositeBlobStore.DEFAULT_STORE_DEFAULT_ID.equals(id)) {
+                    throw new ConfigurationException(CompositeBlobStore.DEFAULT_STORE_DEFAULT_ID
+                            + " is a reserved identifier, please don't use it in the configuration");
+                }
+
+                BlobStore store = null;
+                if (enabled) {
+                    store = config.createInstance();
+                }
+
+                LiveStore liveStore = new LiveStore(config, store);
+                stores.put(config.getId(), liveStore);
+
+                if (config.isDefault()) {
+                    if (defaultStore == null) {
+                        if (!enabled) {
+                            throw new ConfigurationException(
+                                    "The default blob store can't be disabled: " + config.getId());
+                        }
+
+                        defaultStore = config;
+                        stores.put(CompositeBlobStore.DEFAULT_STORE_DEFAULT_ID, liveStore);
+                    } else {
+                        throw new ConfigurationException("Duplicate default blob store: "
```

Comment:
It'd be nice from a UI perspective to scan the rest for default stores so we can give a complete list of duplicates rather than just reporting the first two.